### ISSUE TITLE
Parser gap: a command substitution after a bracket expression in a flagged subscript pattern is rejected

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#379]
## Summary

A flagged subscript pattern that has a command substitution, arithmetic expansion, or grave-accent substitution straight after a bracket expression is valid Zsh. The parser rejects it in every context.

The flagged-pattern repair in `internal/parse/subscript_flag_pattern.go` (#358, #367, #371, #368) masks the bracket expression's `]`. The retry still fails, though. The front end cuts the pattern at that `]` and reads `$(` as a parameter-expansion operator, and none of the repair's error-gated arms name that text.

Found while probing #368 (PR #378). This is independent of it: every row below fails the same way on `4c690f7` (`main`) and on the #378 branch.

## Minimized reproduction

Every row passes `zsh -f -n` and was run under zsh 5.9.2 with `m=(a1 abx abc); x=c`.

| Source | Native run | zsh-lint |
| --- | --- | --- |
| `print ${m[(i)a[bc]$(echo x)]}` | `2` | ``1:19: not a valid parameter expansion operator: `$` `` |
| ``print ${m[(i)a[bc]`echo x`]}`` | `2` | ``1:19: not a valid parameter expansion operator: "`" `` |
| `print ${m[(i)a[bc]$(( 1 ))]}` | `4` | `1:19`, same text as the first row |
| `print ${m[(i)[ab]$(echo x)*]}` | `4` | `1:18`, same text |
| `print ${m[(r)a[bc]$(echo c)]}` | `abc` | `1:19`, same text |
| `print "${m[(i)a[bc]$(echo x)]}"` | `2` | `1:20`, same text |
| `m[(i)a[bc]$(echo x)]=1` | assigns | ``1:1: `a[b]` must be followed by `=` `` |

Controls that already parse:

| Source | zsh-lint |
| --- | --- |
| `print ${m[(i)ab$(echo x)]}` (no bracket expression) | ok |
| `print ${m[(i)a[bc]$x]}`, `print ${m[(i)a[bc]${x}]}` (a parameter expansion, not a substitution) | ok |

The same shape in arithmetic, `$(( m[(i)a[bc]$(echo x)] ))`, is also rejected (`` not a valid arithmetic operator: `$(` ``).

## Notes for the fix

- The error-gated arms would need a new text. A positional gate like the one #378 adds works only if the retry can read `$(` inside the masked literal. Check whether `scanFlagPatternBrackets` stops at `$(`, and whether the mask leaves the substitution for the front end to parse.
- A byte scan must not count parentheses inside the substitution as pattern parentheses (compare #280).
- The assignment row is a different entry point (`a[b]` must be followed by `=`) and may need its own adapter.

## Acceptance

- Every row in the first table parses. The first six keep their native `${...}` tree, and the assignment row keeps its assignment.
- `zsh -f -n`–invalid variants stay rejected, such as an unclosed `$(` inside the pattern.
- Add a fixture `ok-flag-pattern-command-substitution.zsh` plus unit rows.

**Comment by ss-o** [comment]
## Root cause (investigation, on `f99faa6`)

The rejection is deliberate, not a missed case. `scanFlagPatternBrackets` in `internal/parse/subscript_flag_pattern.go` refuses any pattern that holds either form of command substitution:

```go
case b == '`', b == '$' && i+1 < len(src) && src[i+1] == '(':
	return nil, 0, false
```

This line was added in 5626903 (#358). The function's doc comment gives the reason: it reports false when the pattern "holds anything whose extent it cannot decide, such as a nested expansion or a command substitution in either form". `${` was later made scannable by brace nesting (#371). `$(` and grave accents were not.

Without the bracket expression, `${m[(i)ab$(echo x)]}` parses because nothing needs masking: the front end reads the substitution natively. Once there is a `]` to mask, the scanner has to cross the substitution to find the pattern's real end, and it refuses to.

## What a fix needs

1. **Scanner.** Skip a `$( ... )` or `` ` ... ` `` span without masking inside it. Its end must be found with quote awareness: a `)` inside quotes, inside a nested `$(`, or in a `case` pattern inside the substitution does not close it. This is the same byte-count-versus-quote trap as #280 and the #371 lesson. The narrowest safe version refuses a substitution that contains a quote, `case`, a here-document, or a nested substitution, and handles only the flat form.
2. **Verification.** `holdsFlagPattern` requires the pattern to be one `*syntax.Lit` spanning `[start, end)`. With a substitution the pattern is a multi-part word (`Lit`, `CmdSubst`, `Lit`), so the check must accept a word whose parts together span `[start, end)`. `restorePatternEdits` must restore only the literal parts and never touch bytes inside the substitution.
3. **`$(( ))` inside the pattern** (`a[bc]$(( 1 ))`) is arithmetic expansion, not command substitution. Its extent is set by parenthesis depth, and the same quote rules apply.
4. **The assignment row**, `m[(i)a[bc]$(echo x)]=1`, enters through the assignment arm (`flagPatternAfterAssignName`). It is fixed by the same scanner change and needs its own unit row.

## Oracles for the fix

- `zsh -f -n` for validity. Rows with an unclosed `$(` inside the pattern must stay rejected.
- A runtime run with a no-substitution control, to separate runtime failures of the construct from ones that belong to the pattern itself.
- Include a probe grid over parameter-expansion, arithmetic, assignment and double-quoted contexts, since the scanner is shared by every arm of the repair.
